sdk/src/prover.rs

```diff
@@ -17,23 +17,14 @@ use std::time::Duration;
 use alloy_primitives::FixedBytes;
 use anyhow::{Context, Result};
 use bonsai_sdk::alpha as bonsai_sdk;
-use risc0_zkvm::{
-    compute_image_id, default_executor, is_dev_mode, serde::to_vec, ExecutorEnv, Receipt,
-};
+use risc0_zkvm::{compute_image_id, default_executor, is_dev_mode, ExecutorEnv, Receipt};
 
 use crate::snark::{Proof, Seal};
 
-/// Serializes the given input as a `Vec<u8>` compatible with the zkVM and
-/// Bonsai.
-pub fn serialize<T: serde::Serialize + Sized>(input: T) -> Result<Vec<u8>> {
-    let input_data = to_vec(&input)?;
-    Ok(bytemuck::cast_slice(&input_data).to_vec())
-}
-
 /// Generates a snark proof for the given elf and input.
 /// When `RISC0_DEV_MODE` is set, executes the elf locally,
 /// as opposed to sending the proof request to the Bonsai service.
-pub fn generate_proof(elf: &[u8], input: Vec<u8>) -> Result<Proof> {
+pub(crate) fn generate_proof(elf: &[u8], input: impl serde::Serialize) -> Result<Proof> {
```

Comment:
So I think this function should be exposed by the ethereum-sdk; people might need it outside of the starter-template

Reply:
I don't think the utilities provided are ready for wider use as a public API, and so I am hesitant to encourage folks to use this as a library. I'd like to keep types like `Proof` internal for now if possible.

Reply:
My take on the current maturity of this code is that users should use it as an example, and copy it if it solves their problem. And we'll aim to move the into risc0 or the new risc0-ethereum repo soon.